src/components/chapters/segments-wrapper.js

```diff
@@ -55,7 +55,7 @@ class SegmentsWrapper extends Component {
   componentWillUpdate(): void {
     if (!this._lastSegmentUpdated) {
```

Comment:
remind us why we need update last segment endtime?

Reply:
so i couldn't remember exactly why i added it, however, i do remember that it was related to the fact that sources.duration and props.duration are not the same values and it affected on the calculation of buffered/progress and if i remember correctly it also caused like a weird behavior when playing the end of the video. it is more noticeable in short entries.

Reply:
the changes will work but I would rather recommend postpone add chapters on timeline (add segments) till duration of media gets correct. We can use some event for it (f.e. sourceloaded, etc)